apps/gdalalg_vector_change_field_type.cpp

```diff
@@ -143,46 +143,37 @@ class GDALVectorChangeFieldTypeAlgorithmLayer final
         if (m_poFeatureDefn->GetFieldDefn(m_fieldIndex)->GetType() !=
             m_sourceFieldType)
         {
-            // Store the old value and type in a temporary feature
-            std::unique_ptr<OGRFeatureDefn> poFeatureDefTemp =
-                std::make_unique<OGRFeatureDefn>(m_poFeatureDefn->GetName());
-            poFeatureDefTemp->AddFieldDefn(
-                std::make_unique<OGRFieldDefn>("__dummy__", m_sourceFieldType));
-            OGRFeature oFeatureTemp{poFeatureDefTemp.release()};
-            const int tempMap[1] = {m_fieldIndex};
-            oFeatureTemp.SetFieldsFrom(poSrcFeature.get(), tempMap, false,
-                                       true);
-
-            // Remove the old field value
-            poSrcFeature->UnsetField(m_fieldIndex);
-
-            // Change the type
-            poSrcFeature->SetFDefnUnsafe(m_poFeatureDefn);
-
-            // Convert the value
-            std::vector<int> fieldsMap(m_poFeatureDefn->GetFieldCount());
-            for (int i = 0; i < m_poFeatureDefn->GetFieldCount(); i++)
-            {
-                fieldsMap[i] = i == m_fieldIndex ? 0 : -1;
-            }
-            const auto result{poSrcFeature->SetFieldsFrom(
-                &oFeatureTemp, fieldsMap.data(), false, true)};
+            auto poDstFeature = std::make_unique<OGRFeature>(m_poFeatureDefn);
+            std::vector<int> identityMap(m_poFeatureDefn->GetFieldCount());
+            std::fill(identityMap.begin(), identityMap.end(), -1);
```

Comment:
That doesn't feel correct. It should be such identityMap[i] = i for all i. Otherwise you're going to fill only the field m_fieldIndex.  Slight optimization: that map could actually be a member variable